src/IFeatureManager.ts

```diff
@@ -0,0 +1,26 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { ITargetingContext } from "./common/ITargetingContext";
+import { Variant } from "./variant/Variant";
+
+export interface IFeatureManager {
+    /**
+     * Get the list of feature names.
+     */
+    listFeatureNames(): Promise<string[]>;
+
+    /**
+     * Check if a feature is enabled.
+     * @param featureName name of the feature.
+     * @param context an object providing information that can be used to evaluate whether a feature should be on or off.
+     */
+    isEnabled(featureName: string, context?: unknown): Promise<boolean>;
+
+    /**
+     * Get the allocated variant of a feature given the targeting context.
+     * @param featureName name of the feature.
+     * @param context a targeting context object used to evaluate which variant the user will be assigned.
+     */
+    getVariant(featureName: string, context: ITargetingContext): Promise<Variant | undefined>;

```

Comment:
Given the possibility that we calculate audience context id based on any other property rather than userid, I agree that the context is unknown (or to limit it to an object?). A good thing is, here all properties in ITargetingContext are optional, meaning it’s compatible with an arbitrary object type.